src/Altinn.Platform/Altinn.Platform.Storage/Storage/Controllers/MessageboxInstancesController.cs

```diff
@@ -186,7 +189,7 @@ public async Task<ActionResult> Undelete(int instanceOwnerPartyId, Guid instance
             }
             else if (instance.Status.SoftDeleted.HasValue)
             {               
-                instance.LastChangedBy = User?.Identity?.Name;
+                instance.LastChangedBy = User.GetUserOrOrgId();
```

Comment:
User can never null anymore?

Reply:
I don't think so.. Could you confirm, @dskogan ?

Reply:
User.GetUserOrOrgId() will return a value or null if user does not have claims